### ISSUE TITLE
'Expand' action on server doesn't work until another server is selected

### ISSUE THREAD

**Issue by mcx808** [microsoft/azuredatastudio#18840]
Issue Type: <b>Bug</b>

When trying to expand a server on the servers pane, the expand loading wheel remains indefintely, UNTIL another server is selected. Then the previous server expands, and the newly selected server loads indefinitely again.

Bug seems to have appeared for me in the previous to current version.

Azure Data Studio version: azuredatastudio 1.35.1 (fab63efd307e54c063c8af03474bbb5d77f5fd11, 2022-03-15T18:00:06.749Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|11, 15, 13|
|Memory (System)|16.00GB (0.06GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
agent|Mic|0.48.0
dacpac|Mic|1.8.0
import|Mic|1.5.0
profiler|Mic|0.11.0
query-history|Mic|0.1.0
schema-compare|Mic|1.12.0
sql-assessment|Mic|0.6.1
sql-database-projects|Mic|0.14.1
whoisactive|Mic|0.1.4
extra-sql-script-as|pac|0.3.0
sql-search|Red|0.4.0

</details>

**Comment by kburtram** [comment]
@mcx808 could you please check if there are any errors in the "Help->Developer Tools" Console when you see this behavior?  Or if there's any additional context that could help investigate further?  I wasn't able to repro in my test env.

**Comment by mcx808** [comment]
Only a `warn` level error just as the expansion hangs (servername redacted)

```
workbench.desktop.main.js:841  INFO Creating new connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:SqlLogin|database:|server:[server]|user:DBA|group:eb1c2c28-7a5d-4b6a-8661-97767ffecc38
workbench.desktop.main.js:841  INFO Adding connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:SqlLogin|database:|server:[server]|user:DBA|group:eb1c2c28-7a5d-4b6a-8661-97767ffecc38
workbench.desktop.main.js:841  INFO Successfully added connection connection:providerName:MSSQL|applicationName:azdata|authenticationType:SqlLogin|database:|server:[server]|user:DBA|group:eb1c2c28-7a5d-4b6a-8661-97767ffecc38
workbench.desktop.main.js:841  INFO Adding connection with DB name connection:providerName:MSSQL|applicationName:azdata|authenticationType:SqlLogin|database:master|server:[server]|user:DBA|group:eb1c2c28-7a5d-4b6a-8661-97767ffecc38
workbench.desktop.main.js:841  WARN Cannot find node status for session: [server]_master_DBA_SqlLogin_eb1c2c28-7a5d-4b6a-8661-97767ffecc38 and node path: [server]
```

Strangely enough after performing this test ~10 times I'd say a couple of times the server expansion works straight away, but when it does work I do get an error in the console:

`workbench.desktop.main.js:841   ERR Failed to create Object Explorer session`

I also disabled all extensions, but the issue persists.

**Comment by mcx808** [comment]
@kburtram I saw in release 1.39.1

>Bug fixes in 1.39.1
>Fixed bug that caused Database Trees in server connections to not expand in the Object Explorer.

But this issue is still open, is it the same bug?

**Comment by kburtram** [comment]
@mcx808 the bug fixed in 1.39.1 was a specific regression in the August release that made this issue much more common, with a relatively consistent repro, so we were able to address it quickly.  There are still other persistent bugs in this area that result in this behavior that we've had a hard time reproducing so they have been open for a while.  For the Oct release we're planning to add additional diagnostic logging to this code path to try to better differentiate the various errors that result in this behavior to help resolve these long standing bugs.

**Comment by mrlife** [comment]
I am running into this and seeing these errors in the console:

> common.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND 

> console.ts:137 [Extension Host] (node:29702) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.(Use `Azure Data Studio Helper (Renderer) --trace-deprecation ...` to show where the warning was created)

There are lots of warnings like:

> DevTools failed to load source map: Could not load content for https://sqlopsbuilds.blob.core.windows.net/sourcemaps/... HTTP error: status code 400, net::ERR_HTTP_RESPONSE_CODE_FAILURE

**Comment by mcx808** [comment]
Unfortunately I can report this bug has returned for me in 1.43.0

**Comment by cheenamalhotra** [comment]
Hi @mcx808 

We've shifted to async server tree since last few releases? Is this still happening for you in latest release?
Thanks!

**Comment by mcx808** [comment]
Hi @cheenamalhotra, it looks like the issue is resolved in 1.46.0

Thanks!
